### DISCUSSION TITLE
"Make clobberall" will delete the dynamic library with so suffix in wasp install dir

### DISCUSSION THREAD

**Discussion by Ttw0626** [idaholab/moose#25710] [Q&A General] [answered]
Dear moose-developer,

good afternoon, here is a small problem about the usual operation "Make clobberall".

**1) the new installation need the update_recompile for wasp**

The installation on redhat 9 server following the new installation documentation is smooth

**2) make new application**

this usual operation is needed by the path env for library of wasp, the "update_recompile for wasp" will create the links "*.so".

**3) make clobberall seems it will delete these links "*.so"**

this is to say, before the new application compile, I have to re-run the "update_recompile for wasp" to create the link "*.so" everytime once The "Make clobberall" did ?

**4) I did not try the conda env ever, maybe it will be work**

best regards,
Tianwen

**Comment by milljm** [comment]
If you are referring to these instructions: [Create an Application](https://mooseframework.inl.gov/getting_started/new_users.html#compile) I think you are correct. I don't really see how one would know what to do. Either installing the Conda environment, or running any of the `update_and_rebuild_$$$$` scripts. I'll begin addressing this in: https://github.com/idaholab/moose/issues/25711

If you are referring to another set of instructions, can you supply a link?

As for `make clobberall`, deleting `.so` files is the desired effect. Having those lying around when attempting to build anew can cause linking errors. Basically, `make clobberall` is supposed to perform something similar as to: `git clean -xfd` (in each submodule). Unless I am misunderstanding you?

**Comment by loganharbour** [comment]
I think this is actually a valid issue. We don't clean up dependencies (libmesh, petsc) with `make clobberall`, so we shouldn't be doing it for wasp.

**Reply by milljm** [reply]
Oh I see. Yeah I suppose that does make sense. Re running the `update_and_rebuild` scripts clears those out.

**Comment by loganharbour** [comment, marked as answer]
Created issue https://github.com/idaholab/moose/issues/25717.

@permcody will address this soon. Thanks for the report @Ttw0626.

**Reply by Ttw0626** [reply]
@milljm @loganharbour Thanks for this response, 
yeah, I was following the official installation documentation of moose. 
logan pointed it out what I wanna say, sorry for this misunderstanding.

**After "Make clobberall"**
![error_Make_clobberall](https://github.com/idaholab/moose/assets/90584327/119d33c9-351d-430f-810e-69f9cd4d8f45)

best regards,
tianwen

**Reply by permcody** [reply]
I looked into this and it looks like the current behavior is consistent with existing dependencies. That is, we are clobbering libMesh, PETSc, and WASP if they are built locally with `clobber`. To see this behavior, one just needs to inspect the [clobber.py](https://github.com/idaholab/moose/blob/723a2b6ba9a9919726fe2654f8c9b85cd2c8f808/scripts/clobber.py#L86-L89) script. We use a `os.walk` and remove a few select directories (notably repository meta directories) but walk everything else. 

If you don't like the behavior of `clobberall`, I would recommend using `clean` or `cleanall` instead, which is usually good enough unless you've changed deeper dependencies in your change, which may interfere with stale object suffix names. Another way is to use `git clean` which doesn't traverse into git submodules by default.
